File: backend/src/main/java/ua/com/flowershop/entity/Group.java
package ua.com.flowershop.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

import javax.persistence.*;
import java.util.Set;

@Getter
@Setter
@Accessors(chain = true)
@NoArgsConstructor
@AllArgsConstructor
@Entity()
@Table(name = "groups")
public class Group {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String nameSingle;
    private String nameOriginal;
    private String nameOriginalSingle;

    @ManyToOne
    @JoinColumn(name = "flower_type_id")
    private FlowerType flowerType;

    @OneToMany(mappedBy = "group", fetch = FetchType.LAZY)
    private Set<Flower> flowers;

}

File: frontend/src/app/api/models/Group.ts
import { Flower } from "./Flower";
import { FlowerType } from "./FlowerType";

export class Group {
  id?: number;
  name: string;
  nameSingle: string;
  nameOriginal: string;
  nameOriginalSingle: string;
  description: string;
}

export class GroupAdmin extends Group {
  flowerType: FlowerType;
  flower: Flower;
  flowersCount: number;
}
